nixio/data_frame.py

```diff
@@ -137,10 +140,20 @@ def read_columns(self, index=None, name=None, sl=None):
             slic = np.s_[:]
         else:
             slic = np.s_[sl]
-        get_col = self._read_data(sl=slic)[name]
         if len(name) == 1:
+            get_col = self._read_data(sl=slic)[name]
             get_col = [i[0] for i in get_col]
-        return get_col
+            return get_col
+        if group_by_cols:
+            gcol = []
+            for n in name:
+                get_col = self._read_data(sl=slic)[n]
+                get_col = [i for i in get_col]
+                gcol.append(get_col)
+            return gcol
+        else:
+            get_col = self._read_data(sl=slic)[name]
+            return get_col
```

Comment:
Having a switch for col-major vs row-major ordering is nice, but the two options return different types, which isn't very friendly.

Row-major order (default) returns a numpy.array while col-major order returns a list.

Reply:
agreed